libsrc/target/aquarius/interrupt/im1_init.asm

```diff
@@ -0,0 +1,64 @@
+        SECTION code_clib
+        PUBLIC  _im1_init
+        PUBLIC  im1_init
+
+        EXTERN  asm_interrupt_handler
+        EXTERN  im1_vectors
+        EXTERN  l_pop_ei
+        EXTERN  l_push_di
+        extern  tick_count_isr
+
+        defc    Z80_OPCODE_JP=$c3
+        defc    EXISTING_INT_SIZE=3
+        defc    IO_IRQMASK=$ee
+        defc    IO_IRQSTAT=$ef
+        defc    IO_BANK0=$f0
+	defc	INTJMP=$38fb
+
+im1_init:
+_im1_init:
+        ; Save info at address INTJMP
+        ld      hl, INTJMP
+        ld      de, existing_int
+        ld      bc, EXISTING_INT_SIZE
+        ldir
+
+        ; Write code to jump to our handler
+        ld      a, Z80_OPCODE_JP
+        ld      (INTJMP), a
+        ld      hl, asm_im1_handler
+        ld      (INTJMP+1), hl
+
+        ; Unmask and clear VBLANK IRQ
+        ld      a, $01
+        out     (IO_IRQMASK), a
+        out     (IO_IRQSTAT), a
+
+        ei
```

Comment:
Should there not be a di whilst we're messing with the vectors?

To preserve the previous interrupt state, this is the pattern we use:

```
call l_push_di. ; saves enabled state
.... ; critical code
call l_pop_ei ; restores enabled state
```

Reply:
Hi @suborb , thanks for taking a look. To be safe there absolutely be a di before messing with the vector I will add this.

However, I'm hesitant to use l_push_di and l_pop_ei due to the fact that examples like the sound example call im1_init but do not perform an explicit ei, so my assumption is im1_init enables interrupts on exit. What are your thoughts?

Reply:
That's a good point - yes, an ei would be correct then.